action.yml

```diff
@@ -28,16 +32,44 @@ runs:
         key: dune-cache-${{ runner.os }}-${{ runner.arch }}
     - name: Resolve dependencies
       shell: bash
+      env:
+        SETUPDUNEDIR: ${{ inputs.directory }}
+      run: |
+        cd "$SETUPDUNEDIR" && dune pkg lock
+      if: inputs.automagic == 'true'
+    - name: Install depexts
+      shell: bash
+      env:
+        SETUPDUNEDIR: ${{ inputs.directory }}
+        OS: ${{ runner.os }}
       run: |
```

Comment:
I started doing some reading about embedding shell scripts in YAML actions vs referencing separated shell scripts. My personal threshold for favoring a `shellcheck`-ed script would be very low in term of number of lines, but I take you probably wouldn't like having a linting dependency on `shellcheck`?